packages/core/src/QueryManager.js

```diff
@@ -62,13 +66,46 @@ export class QueryManager {
       }
       const data = await this.db.query({ type, sql, ...options });
       if (cache) this.clientCache.set(sql, data);
-      this._logger.debug(`Request: ${(performance.now() - t0).toFixed(1)}`);
+      this._logger.debug(`Request: ${(performance.now() - t0).toFixed(1)} ms, ${(t0).toFixed(1)} ms`);
       result.fulfill(data);
     } catch (err) {
       result.reject(err);
     }
   }
 
+  async submitBatch(requests, results) {
+    for(let i = 0; i < requests.length; i++){
+      try {
+        const { query, type, cache = false, record = true, options } = requests[i];
+        const sql = query ? `${query}` : null;
+        if (record) {
+          this.recordQuery(sql);
+        }
+        if (cache) {
+          const cached = this.clientCache.get(sql);
+          if (cached) {
+            this._logger.debug('Cache');
+            results[i].fulfill(cached);
+            continue;
+          }
+        }
+        const t0 = performance.now();
+        if (this._logQueries) {
+          this._logger.debug('Query', { type, sql, ...options });
+        }
+        this.db.query({ type, sql, ...options }).then(data => {
+          if (cache) this.clientCache.set(sql, data);
+          this._logger.debug(`Request: ${(performance.now() - t0).toFixed(1)}`);
+          results[i].fulfill(data);
```

Comment:
I don't think this correctly sorts the results yet, no? The request is fulfilled in the order of the `then` in line 96.